src/Propel/Runtime/ActiveQuery/QueryExecutor/InsertQueryExecutor.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * MIT License. This file is part of the Propel package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Propel\Runtime\ActiveQuery\QueryExecutor;
+
+use Propel\Runtime\ActiveQuery\Criteria;
+use Propel\Runtime\ActiveQuery\SqlBuilder\InsertQuerySqlBuilder;
+use Propel\Runtime\Connection\ConnectionInterface;
+use Propel\Runtime\Exception\PropelException;
+use Throwable;
+
+class InsertQueryExecutor extends AbstractQueryExecutor
+{
+    /**
+     * @var \Propel\Runtime\Map\ColumnMap|null
+     */
+    protected $primaryKeyColumn;
+
+    /**
+     * @param \Propel\Runtime\ActiveQuery\Criteria $criteria
+     * @param \Propel\Runtime\Connection\ConnectionInterface|null $con
+     */
+    protected function __construct(Criteria $criteria, ?ConnectionInterface $con = null)
+    {
+        parent::__construct($criteria, $con);
+
+        $this->primaryKeyColumn = $criteria->getPrimaryKey();
+    }
+
+    /**
+     * @see InsertQuerySqlBuilder::runInsert()
+     *
+     * @param \Propel\Runtime\ActiveQuery\Criteria $criteria
+     * @param \Propel\Runtime\Connection\ConnectionInterface|null $con
+     *
+     * @return mixed|null
```

Comment:
is it really mixed? isnt it only `string|int|null`?

Reply:
I think you are right, but the return chain ends with AdapterInterface::getId(), where return type is set to `mixed`
```php
/**
     * Gets the generated ID (either last ID for autoincrement or next sequence ID).
     *
     * @param \Propel\Runtime\Connection\ConnectionInterface $con
     * @param string|null $name
     *
     * @return mixed
     */
    public function getId(ConnectionInterface $con, $name = null);
```
I don't think I can change this, can I?

Reply:
We can change the docblock (and the actual truth of values expected to be returned). This will not have any consequences on the code itself.

Reply:
I'll be damned, stan and psalm don't complain. Nice!